## Analysis

In [1]:
# Dependencies and Setup
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

In [2]:
# Study data files
AMAZON_PATH = "../Resources/amazonCopy.csv"

In [3]:
# Read the Amazon data and the study results
Amazon_data = pd.read_csv(AMAZON_PATH, encoding="utf-8")

In [4]:
#checking our dataset
Amazon_data.head(5)

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             ₹399   
1  Computers&Accessories|Accessories&Peripherals|...             ₹199   
2  Computers&Accessories|Accessories&Peripherals|...             ₹199   
3  Computers&Accessories|Accessories&Peripherals|...             ₹329   
4  Computers&Accessories|Accessories&Peripherals|...             ₹154   

  actual_price discount_percentage rating rating_count  \
0       ₹1,099                 64%    4.2       24,269   
1         ₹349                 43%      4       43,994   
2       ₹1,899                 90%    3.9        7,928   
3         ₹699                 53%    4.2       94,363   
4         ₹399                 61%    4.2       16,905   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                           review_id  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...   
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...   
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...   
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...   

                                        review_title  \
0  Satisfied,Charging is really fast,Value for mo...   
1  A Good Braided Cable for Your Type C Device,Go...   
2  Good speed for earlier versions,Good Product,W...   
3  Good product,Good one,Nice,Really nice product...   
4  As good as original,Decent,Good one for second...   

                                      review_content  \
0  Looks durable Charging is fine tooNo complains...   
1  I ordered this cable to connect my phone to An...   
2  Not quite durable and sturdy,https://m.media-a...   
3  Good product,long wire,Charges good,Nice,I bou...   
4  Bought this instead of original apple, does th...   

                                            img_link  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   
1  https://m.media-amazon.com/images/W/WEBP_40237...   
2  https://m.media-amazon.com/images/W/WEBP_40237...   
3  https://m.media-amazon.com/images/I/41V5FtEWPk...   
4  https://m.media-amazon.com/images/W/WEBP_40237...   

                                        product_link  
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  
1  https://www.amazon.in/Ambrane-Unbreakable-Char...  
2  https://www.amazon.in/Sounce-iPhone-Charging-C...  
3  https://www.amazon.in/Deuce-300-Resistant-Tang...  
4  https://www.amazon.in/Portronics-Konnect-POR-1...

In [5]:
Amazon_data.shape

(1465, 16)

In [6]:
Amazon_data.nunique()

product_id             1351
product_name           1337
category                211
discounted_price        550
actual_price            449
discount_percentage      92
rating                   26
rating_count           1143
about_product          1293
user_id                1194
user_name              1194
review_id              1194
review_title           1194
review_content         1212
img_link               1412
product_link           1465
dtype: int64

In [7]:
Amazon_data.dtypes

product_id             object
product_name           object
category               object
discounted_price       object
actual_price           object
discount_percentage    object
rating                 object
rating_count           object
about_product          object
user_id                object
user_name              object
review_id              object
review_title           object
review_content         object
img_link               object
product_link           object
dtype: object

In [8]:
Amazon_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   product_id           1465 non-null   object
 1   product_name         1465 non-null   object
 2   category             1465 non-null   object
 3   discounted_price     1465 non-null   object
 4   actual_price         1465 non-null   object
 5   discount_percentage  1465 non-null   object
 6   rating               1465 non-null   object
 7   rating_count         1463 non-null   object
 8   about_product        1465 non-null   object
 9   user_id              1465 non-null   object
 10  user_name            1465 non-null   object
 11  review_id            1465 non-null   object
 12  review_title         1465 non-null   object
 13  review_content       1465 non-null   object
 14  img_link             1465 non-null   object
 15  product_link         1465 non-null   object
dtypes: obj

In [9]:
# changing the data types of discounted_price, actual_price, discount_percentage, and rating_count.
#Amazon_data["discounted_price"] = pd.to_numeric(Amazon_data["discounted_price"])
# First Remove the Rupee symbol and any other non-numeric characters except the decimal point
Amazon_data["discounted_price"] = Amazon_data["discounted_price"].str.replace('₹', '').str.replace(',', '')
Amazon_data["actual_price"] = Amazon_data["actual_price"].str.replace('₹', '').str.replace(',', '')
Amazon_data["rating_count"] = Amazon_data["rating_count"].str.replace(',', '')
Amazon_data["rating"] = Amazon_data["rating"].str.replace(',', '')


# Convert the columns to numeric, coercing errors to NaN (in case there are any non-numeric values left)
Amazon_data["discounted_price"] = pd.to_numeric(Amazon_data["discounted_price"], errors='coerce')
Amazon_data["actual_price"] = pd.to_numeric(Amazon_data["actual_price"], errors='coerce')
Amazon_data["rating_count"] = pd.to_numeric(Amazon_data["rating_count"], errors='coerce')
Amazon_data["rating"] = pd.to_numeric(Amazon_data["rating"], errors='coerce')

In [10]:
# Remove the percent sign from the 'discount_percentage' column
Amazon_data['discount_percentage'] = Amazon_data['discount_percentage'].str.replace('%', '')

# Convert the 'discount_percentage' column to numeric, coercing errors to NaN
Amazon_data['discount_percentage'] = pd.to_numeric(Amazon_data['discount_percentage'], errors='coerce')

# convert the percentages to a fraction 
Amazon_data['discount_percentage'] = Amazon_data['discount_percentage'] / 100


In [11]:
Amazon_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   product_id           1465 non-null   object 
 1   product_name         1465 non-null   object 
 2   category             1465 non-null   object 
 3   discounted_price     1465 non-null   float64
 4   actual_price         1465 non-null   float64
 5   discount_percentage  1465 non-null   float64
 6   rating               1464 non-null   float64
 7   rating_count         1463 non-null   float64
 8   about_product        1465 non-null   object 
 9   user_id              1465 non-null   object 
 10  user_name            1465 non-null   object 
 11  review_id            1465 non-null   object 
 12  review_title         1465 non-null   object 
 13  review_content       1465 non-null   object 
 14  img_link             1465 non-null   object 
 15  product_link         1465 non-null   o

In [12]:
#count the number of missing values (NaNs) in each column of the DataFrame 
Amazon_data.isna().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 1
rating_count           2
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

In [13]:
# removing any rows in the DataFrame Amazon_data that contain missing values (NaNs) across any column
#checking the dataset again
Amazon_data.dropna(inplace = True)
print(Amazon_data.shape)
Amazon_data.isna().sum()

(1462, 16)


product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           0
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

In [14]:
#converting discounted_price, rating, and actual price to integer
Amazon_data["discounted_price"] = Amazon_data["discounted_price"].astype(int)
Amazon_data["actual_price"] = Amazon_data["actual_price"].astype(int)
Amazon_data["rating"] = Amazon_data["rating"].astype(int)

In [15]:
Amazon_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1462 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   product_id           1462 non-null   object 
 1   product_name         1462 non-null   object 
 2   category             1462 non-null   object 
 3   discounted_price     1462 non-null   int32  
 4   actual_price         1462 non-null   int32  
 5   discount_percentage  1462 non-null   float64
 6   rating               1462 non-null   int32  
 7   rating_count         1462 non-null   float64
 8   about_product        1462 non-null   object 
 9   user_id              1462 non-null   object 
 10  user_name            1462 non-null   object 
 11  review_id            1462 non-null   object 
 12  review_title         1462 non-null   object 
 13  review_content       1462 non-null   object 
 14  img_link             1462 non-null   object 
 15  product_link         1462 non-null   object

In [16]:
Amazon_data.nunique()

product_id             1348
product_name           1334
category                211
discounted_price        549
actual_price            449
discount_percentage      92
rating                    4
rating_count           1142
about_product          1290
user_id                1191
user_name              1191
review_id              1191
review_title           1191
review_content         1209
img_link               1409
product_link           1462
dtype: int64

In [17]:
# removing duplicate rows from a DataFrame 
Amazon_data.drop_duplicates(inplace =True)

In [18]:
#checking the dataset again
Amazon_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1462 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   product_id           1462 non-null   object 
 1   product_name         1462 non-null   object 
 2   category             1462 non-null   object 
 3   discounted_price     1462 non-null   int32  
 4   actual_price         1462 non-null   int32  
 5   discount_percentage  1462 non-null   float64
 6   rating               1462 non-null   int32  
 7   rating_count         1462 non-null   float64
 8   about_product        1462 non-null   object 
 9   user_id              1462 non-null   object 
 10  user_name            1462 non-null   object 
 11  review_id            1462 non-null   object 
 12  review_title         1462 non-null   object 
 13  review_content       1462 non-null   object 
 14  img_link             1462 non-null   object 
 15  product_link         1462 non-null   object

In [19]:
#checking the category column
Amazon_data["category"].head(5)

0    Computers&Accessories|Accessories&Peripherals|...
1    Computers&Accessories|Accessories&Peripherals|...
2    Computers&Accessories|Accessories&Peripherals|...
3    Computers&Accessories|Accessories&Peripherals|...
4    Computers&Accessories|Accessories&Peripherals|...
Name: category, dtype: object

In [20]:
#Removing unneccesary strings ('|','&'') from 'category' column for the queries to make in MongoDB file
Amazon_data["category"] = Amazon_data["category"].str.replace('|', '').str.replace('&', '')

In [21]:
#checking the output
Amazon_data["category"].head(5)

0    ComputersAccessoriesAccessoriesPeripheralsCabl...
1    ComputersAccessoriesAccessoriesPeripheralsCabl...
2    ComputersAccessoriesAccessoriesPeripheralsCabl...
3    ComputersAccessoriesAccessoriesPeripheralsCabl...
4    ComputersAccessoriesAccessoriesPeripheralsCabl...
Name: category, dtype: object

In [22]:
#exploring some columns in the dataset
Amazon_data["category"].nunique()

211

In [23]:
Amazon_data["product_name"].nunique()

1334

In [24]:
Amazon_data["user_id"].nunique()

1191

In [25]:
# Count the number of characters in each row of all columns in the dataframe
max_length1 = Amazon_data["product_id"].apply(len).max()
max_length1

10

In [26]:
max_length2 = Amazon_data["product_name"].apply(len).max()
max_length2

485

In [27]:
max_length3 = Amazon_data["category"].apply(len).max()
max_length3

110

In [28]:
max_length4 = Amazon_data["about_product"].apply(len).max()
max_length4

2646

In [29]:
max_length5 = Amazon_data["user_id"].apply(len).max()
max_length5

231

In [30]:
max_length6 = Amazon_data["user_name"].apply(len).max()
max_length6

137

In [31]:
max_length7 = Amazon_data["review_id"].apply(len).max()
max_length7

119

In [32]:
max_length8 = Amazon_data["review_title"].apply(len).max()
max_length8

407

In [33]:
max_length9 = Amazon_data["review_content"].apply(len).max()
max_length9

18547

In [34]:
max_length10 = Amazon_data["img_link"].apply(len).max()
max_length10

114

In [35]:
max_length11 = Amazon_data["product_link"].apply(len).max()
max_length11

188

In [36]:
#checking the columns
Amazon_data.columns

Index(['product_id', 'product_name', 'category', 'discounted_price',
       'actual_price', 'discount_percentage', 'rating', 'rating_count',
       'about_product', 'user_id', 'user_name', 'review_id', 'review_title',
       'review_content', 'img_link', 'product_link'],
      dtype='object')

In [37]:
#Find the maximum  rating counts
max_counts = Amazon_data["rating_count"].max()
max_counts

426973.0

In [38]:
max_value_index = Amazon_data['rating_count'].idxmax()
max_value_index

12

In [39]:
row_with_highest_value = Amazon_data.loc[max_value_index]

print("Row with the highest value:")
print(row_with_highest_value)

Row with the highest value:
product_id                                                    B07KSMBL2H
product_name           AmazonBasics Flexible Premium HDMI Cable (Blac...
category               ElectronicsHomeTheater,TVVideoAccessoriesCable...
discounted_price                                                     219
actual_price                                                         700
discount_percentage                                                 0.69
rating                                                                 4
rating_count                                                    426973.0
about_product          Flexible, lightweight HDMI cable for connectin...
user_id                AEYJ5I6JZZPOJB6MGWRQOHRQLPSQ,AFY5TVFOMVHGBPBTI...
user_name              Rishav Gossain,Shashank Mallamraju,Purnendu,gu...
review_id              R1FKOKZ3HHKJBZ,R2WNMZI1EXTA0H,RCA1M3W4RIXUR,R3...
review_title           It's quite good and value for money,Works well...
review_content         

In [40]:
#Checking the dataset again
Amazon_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1462 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   product_id           1462 non-null   object 
 1   product_name         1462 non-null   object 
 2   category             1462 non-null   object 
 3   discounted_price     1462 non-null   int32  
 4   actual_price         1462 non-null   int32  
 5   discount_percentage  1462 non-null   float64
 6   rating               1462 non-null   int32  
 7   rating_count         1462 non-null   float64
 8   about_product        1462 non-null   object 
 9   user_id              1462 non-null   object 
 10  user_name            1462 non-null   object 
 11  review_id            1462 non-null   object 
 12  review_title         1462 non-null   object 
 13  review_content       1462 non-null   object 
 14  img_link             1462 non-null   object 
 15  product_link         1462 non-null   object

In [41]:
#checking the unique column names in the 'category' column
Amazon_data_unique_category = Amazon_data["category"].unique()
Amazon_data_unique_category

array(['ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesUSBCables',
       'ComputersAccessoriesNetworkingDevicesNetworkAdaptersWirelessUSBAdapters',
       'ElectronicsHomeTheater,TVVideoAccessoriesCablesHDMICables',
       'ElectronicsHomeTheater,TVVideoTelevisionsSmartTelevisions',
       'ElectronicsHomeTheater,TVVideoAccessoriesRemoteControls',
       'ElectronicsHomeTheater,TVVideoTelevisionsStandardTelevisions',
       'ElectronicsHomeTheater,TVVideoAccessoriesTVMounts,StandsTurntablesTVWallCeilingMounts',
       'ElectronicsHomeTheater,TVVideoAccessoriesCablesRCACables',
       'ElectronicsHomeAudioAccessoriesSpeakerAccessoriesMounts',
       'ElectronicsHomeTheater,TVVideoAccessoriesCablesOpticalCables',
       'ElectronicsHomeTheater,TVVideoProjectors',
       'ElectronicsHomeAudioAccessoriesAdapters',
       'ElectronicsHomeTheater,TVVideoSatelliteEquipmentSatelliteReceivers',
       'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesDVICables

In [42]:
# Specify the special word 
special_word = 'Cables'

# Use the str.contains() method to filter the 'category' column for rows containing the special word 'Cables'
# Note: This will return a boolean series
contains_special_word = Amazon_data['category'].str.contains(special_word, na=False)

# Get unique values from the column that contain the special word
unique_values_with_special_word = Amazon_data.loc[contains_special_word, 'category'].unique()

# Print the unique values that contain the special word
print(unique_values_with_special_word)

# checking the unique values
print(f"Number of unique values in 'column_name' containing '{special_word}': {len(unique_values_with_special_word)}")

['ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesUSBCables'
 'ElectronicsHomeTheater,TVVideoAccessoriesCablesHDMICables'
 'ElectronicsHomeTheater,TVVideoAccessoriesCablesRCACables'
 'ElectronicsHomeTheater,TVVideoAccessoriesCablesOpticalCables'
 'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesDVICables'
 'ElectronicsHomeTheater,TVVideoAccessoriesCablesSpeakerCables'
 'ElectronicsMobilesAccessoriesMobileAccessoriesCablesAdaptersOTGAdapters'
 'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCableConnectionProtectors'
 'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesEthernetCables'
 'ComputersAccessoriesAccessoriesPeripheralsCablesAccessoriesCablesSATACables']
Number of unique values in 'column_name' containing 'Cables': 10


In [43]:
# remove the duplicated row in 'product_id'
Amazon_data_clean = Amazon_data.drop_duplicates(subset=["product_id"])

In [44]:
#checking the dataframe 
Amazon_data_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1348 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   product_id           1348 non-null   object 
 1   product_name         1348 non-null   object 
 2   category             1348 non-null   object 
 3   discounted_price     1348 non-null   int32  
 4   actual_price         1348 non-null   int32  
 5   discount_percentage  1348 non-null   float64
 6   rating               1348 non-null   int32  
 7   rating_count         1348 non-null   float64
 8   about_product        1348 non-null   object 
 9   user_id              1348 non-null   object 
 10  user_name            1348 non-null   object 
 11  review_id            1348 non-null   object 
 12  review_title         1348 non-null   object 
 13  review_content       1348 non-null   object 
 14  img_link             1348 non-null   object 
 15  product_link         1348 non-null   object

In [45]:
# Let's say we preemptively replace some special characters with spaces before removing all special characters
Amazon_data_clean['review_title'] = Amazon_data_clean['review_title'].str.replace('-', ' ', regex=False)

# Now, remove all non-alphanumeric characters except spaces
Amazon_data_clean['review_title'] = Amazon_data_clean['review_title'].str.replace('[^a-zA-Z0-9 ]', '', regex=True)

# Capitalize the first letter of each title
Amazon_data_clean['review_title'] =Amazon_data_clean['review_title'].str.capitalize()

# Display the DataFrame to verify the change(Amazon_data_clean)
Amazon_data_clean.head()

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category  discounted_price  \
0  ComputersAccessoriesAccessoriesPeripheralsCabl...               399   
1  ComputersAccessoriesAccessoriesPeripheralsCabl...               199   
2  ComputersAccessoriesAccessoriesPeripheralsCabl...               199   
3  ComputersAccessoriesAccessoriesPeripheralsCabl...               329   
4  ComputersAccessoriesAccessoriesPeripheralsCabl...               154   

   actual_price  discount_percentage  rating  rating_count  \
0          1099                 0.64       4       24269.0   
1           349                 0.43       4       43994.0   
2          1899                 0.90       3        7928.0   
3           699                 0.53       4       94363.0   
4           399                 0.61       4       16905.0   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                           review_id  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...   
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...   
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...   
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...   

                                        review_title  \
0  Satisfiedcharging is really fastvalue for mone...   
1  A good braided cable for your type c devicegoo...   
2  Good speed for earlier versionsgood productwor...   
3  Good productgood onenicereally nice productver...   
4  As good as originaldecentgood one for secondar...   

                                      review_content  \
0  Looks durable Charging is fine tooNo complains...   
1  I ordered this cable to connect my phone to An...   
2  Not quite durable and sturdy,https://m.media-a...   
3  Good product,long wire,Charges good,Nice,I bou...   
4  Bought this instead of original apple, does th...   

                                            img_link  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   
1  https://m.media-amazon.com/images/W/WEBP_40237...   
2  https://m.media-amazon.com/images/W/WEBP_40237...   
3  https://m.media-amazon.com/images/I/41V5FtEWPk...   
4  https://m.media-amazon.com/images/W/WEBP_40237...   

                                        product_link  
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  
1  https://www.amazon.in/Ambrane-Unbreakable-Char...  
2  https://www.amazon.in/Sounce-iPhone-Charging-C...  
3  https://www.amazon.in/Deuce-300-Resistant-Tang...  
4  https://www.amazon.in/Portronics-Konnect-POR-1...

### Analysis

From the above filtering function of the dataset, we retrived 10 unique values (category name) from the 'category' column. Based on the retrived names, we found 7th and 8th outputs are not 'Cables', hence we will only consider 8 unique values being the the cables category in the 'category' column and ignore the other 2 retreived columns which are not cables, which we will use for our further query functions processes in our data exploration inside the 'Mongo_db_analysis.ipynb'.

Likewise we also deleted the duplicated rows on "product_id" column for the proper data analysis process with the resulting 1348 rows and 16 columns left in our dataframe. Finally, we saved our new 'Amazon_data_clean' dataframe naming as **amazon_cleanData** inside the clean data resource folder.

In [46]:
#Finally saving the clean dataset in the Clean data resources folder
Amazon_data_clean.to_csv('../Clean_Data_Resources/amazon_cleanData.csv', index=False, encoding='utf-8')
